junit-jupiter-api/src/main/java/org/junit/jupiter/api/EnabledIf.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apiguardian.api.API;
+
+/**
+ * {@code @EnabledIf} evaluates a script controlling that the annotated
+ * test class or test method is currently <em>enabled</em> or <em>disabled</em>.
+ *
+ * <p>The decision is made by interpreting the return value of the {@link #value() script}:
+ * <ul>
+ * <li>{@code true} - if and only if the String-representation of the returned
+ * value is parsed by {@link Boolean#parseBoolean(String)} to {@code true}.</li>
+ * <li>{@code ConditionEvaluationResult} - an instance of
+ *  {@link org.junit.jupiter.api.extension.ConditionEvaluationResult ConditionEvaluationResult}
+ *  is passed directly to the Jupiter engine.</li>
+ *  </ul>
+ *
+ * <p>When this annotation with a script that evaluates to {@code false}
+ * is applied at the class level, all test methods within that class
+ * are automatically disabled as well.
+ *
+ * <h3>Script engine</h3>
+ * The default script engine is: Oracle Nashorn.
+ *
+ * The {@link #engine()} property overrides the default script engine name.
+ *
+ * <h3>Bindings</h3>
+ * An {@link Accessor accessor} value provides access to a map-like structure.
+ *
+ * <p>The following System property accessors are available:
+ * <ul>
+ * <li>{@link Bind#SYSTEM_ENVIRONMENT systemEnvironment} - System environment accessor.</li>
+ * <li>{@link Bind#SYSTEM_PROPERTY systemProperty} - System property accessor.</li>
+ * </ul>
+ *
+ * The following Jupiter extension context-aware {@link Bind bindings} are available:
+ * <ul>
+ * <li>{@link Bind#JUPITER_TAGS jupiterTags} - All tags as a {@code Set<String>}.</li>
+ * <li>{@link Bind#JUPITER_DISPLAY_NAME jupiterDisplayName} - Display name as a {@code String}.</li>
+ * <li>{@link Bind#JUPITER_UNIQUE_ID jupiterUniqueId} - Unique ID as a {@code String}.</li>
+ * <li>{@link Bind#JUPITER_CONFIGURATION_PARAMETER jupiterConfigurationParameter} - Configuration parameter accessor.</li>
+ * </ul>
+ *
+ * @since 5.1
+ * @see org.junit.jupiter.api.extension.ExecutionCondition
+ * @see org.junit.jupiter.api.extension.ConditionEvaluationResult#enabled(String)
+ * @see org.junit.jupiter.api.extension.ConditionEvaluationResult#disabled(String)
```

Comment:
Add `@see ScriptEngine`?

Reply:
All documentation polishing is targetet to **5.1-RC1** ;)

Reply:
Done.